src/bokeh/models/formatters.py

```diff
@@ -636,7 +637,14 @@ def __init__(self, *args, **kwargs) -> None:
     years = String(help=_DATETIME_TICK_FORMATTER_HELP("``years``"),
                    default="%Y").accepts(List(String), _deprecated_datetime_list_format)
 
-    strip_leading_zeros = Bool(default=True, help="Whether to strip any leading zeros in the formatted ticks")
+    strip_leading_zeros = Either(Bool, List(Enum(ResolutionType)), default=False, help="""
+    Whether to strip any leading zeros in the formatted ticks.
+
+    Valid values are:
+    * ``True`` or ``False`` (default) to set stripping across all resolutions.
+    * A list of resolution types, e.g. ``["microseconds", "milliseconds"]``, to enable
+    scale-dependent stripping of leading zeros.
```

Comment:
Would `Sequence` property work here? If I was personally writing code, I would aways reach for tuples before lists.

Reply:
@bryevdv Sequence property should be doable here as mutability is not required. I will try to provide a commit for that the next days.